### PR TITLE
Add room member

### PR DESCRIPTION
**Pull request by su-chang**
## Checklist

- [ ] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added
- [ ] CI has been passed. (GitHub actions all turns green)
- [ ] SLA has been signed

## Description

> please describe the changes that you are making, with the related issue number.

### CODE REVIEW

File: src/user/friendship.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ import {
 }                   from '../helper-functions/mod'
 
 import {
+  FriendshipAddOptions,
```
Comment (huan) [inline src/user/friendship.ts:34]: `FriendshipAddOptions as PuppetFriendshipAddOptions`

File: src/user/friendship.ts [hunk 2]
```diff
@@ -44,6 +45,16 @@ import {
   Contact,
 }                   from './contact'
 
+import {
+  Room,
+}                   from './room'
+
+interface FriendshipAddOptionsObj {
```
Comment (huan) [inline src/user/friendship.ts:52]: FriendshipAddOptionsObj -> FriendshipAddOptions

File: src/user/friendship.ts [hunk 3]
```diff
@@ -112,24 +123,41 @@ class Friendship extends EventEmitter implements Acceptable {
...
+    option  : string | FriendshipAddOptionsObj,
   ): Promise<void> {
     log.verbose('Friendship', 'static add(%s, %s)',
       contact.id,
-      hello,
+      typeof option === 'string' ? option : option.hello,
     )
-    await this.wechaty.puppet.friendshipAdd(contact.id, hello)
+
+    if (typeof option === 'string') {
```
Comment (huan) [inline src/user/friendship.ts:151]:
```ts
if (typeof options === 'string') {
  await this.wechaty.puppet.friendshipAdd(contact.id, { hello: options })
}
```

File: src/user/friendship.ts [hunk 4]
```diff
@@ -112,24 +123,41 @@ class Friendship extends EventEmitter implements Acceptable {
...
+      await this.wechaty.puppet.friendshipAdd(contact.id, option)
+    } else {
+      const friendOption: FriendshipAddOptions = {
+        hello: option.hello,
+        contactId: option.contact && option.contact.id,
```
Comment (huan) [inline src/user/friendship.ts:156]: `contactId: option?.contact?.id,`

**Review by huan — CHANGES_REQUESTED** [review]
It will be good if we can remove the `string` type in the Puppet definition.

**Comment by su-chang** [comment]
> It will be good if we can remove the `string` type in the Puppet definition.

I think it should be removed after all puppet has updated.

**Comment by windmemory** [comment]
> It will be good if we can remove the `string` type in the Puppet definition.

@huan How about we print out a `warn` log with deprecate messages in it, then remove the `string` type in the next version? Thus this will be a smooth transition for our developres.

**Comment by huan** [comment]
Thanks for the suggestion, I'm ok with that.

Please go ahead and I will modify it as I need after you approved it.

**Review by windmemory — APPROVED** [review]

**Review by huan — CHANGES_REQUESTED** [review]
Thanks for the Approvement!

Please:

1. Bump the minor version number, whenever the core dependencies (like wechaty-puppet, wechaty-puppet-service, etc) have a minor version bumping.
2. Please make the CI green, or let me know if you have any problems need help.

**Comment by su-chang** [comment]
Related issue: https://github.com/wechaty/wechaty/issues/2142

**Comment by su-chang** [comment]
Merge this feature to master branch.

**Review by huan — APPROVED** [review]
LGTM

**Pull request merged**